src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+            };
+
+            Dictionary<string, OpenApiSchema> responseBodyProperties = new()
+            {
+                {
+                    RESPONSE_VALUE_PROPERTY,
+                    responseRootSchema
+                }
+            };
+
+            OpenApiMediaType responsePayload = new()
+            {
+                Schema = new()
+                {
+                    Properties = responseBodyProperties
+                }
+            };
+
+            return responsePayload;
+        }
+
+        /// <summary>
+        /// Builds the schema objects for all entities present in the runtime configuration.
+        /// Two schemas per entity are created:
+        /// 1) {EntityName}      -> Primary keys present in schema, used for request bodies (excluding GET) and all response bodies.
+        /// 2) {EntityName}_NoPK -> No primary keys present in schema, used for POST requests where PK is autogenerated.
+        /// Schema objects can be referenced elsewhere in the OpenAPI document with the intent to reduce document verbosity.
+        /// </summary>
+        /// <returns>Collection of schemas for entities defined in the runtime configuration.</returns>
+        private Dictionary<string, OpenApiSchema> CreateComponentSchemas()
+        {
+            Dictionary<string, OpenApiSchema> schemas = new();
+
+            foreach (string entityName in _metadataProvider.EntityToDatabaseObject.Keys.ToList())
+            {
+                // Entities which disable their REST endpoint must not be included in
+                // the OpenAPI description document.
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+                List<string> columns = /*sourceDefinition is null ? new List<string>() : */sourceDefinition.Columns.Keys.ToList();
+
+                // create component for FULL entity with PK.
+                schemas.Add(entityName, CreateComponentSchema(entityName, fields: columns));
+
+                // create component for entity with no PK
+                // get list of columns - primary key columns then optimize
+                foreach (string primaryKeyColumn in sourceDefinition.PrimaryKey)
+                {
+                    columns.Remove(primaryKeyColumn);
+                }
+
+                // create component for TABLE (view?) NOT STOREDPROC entity with no PK.
+                DatabaseObject dbo = _metadataProvider.EntityToDatabaseObject[entityName];
+                if (dbo.SourceType is not SourceType.StoredProcedure or SourceType.View)
```

Comment:
For storedproc or view, will we always have a schema with PK?

Reply:
Are we not creating schema without PK for view?

Reply:
Updated. Views and Tables will have the same schema objects created due to expected same behavior. Stored Procedures will contain a schema with PK because we require SP params to be included in the request body only. However for GET requests, SP Params can be in the query string. Query string documentation will come in another release/later PR.